pkg/controller/combo_controller.go

```diff
@@ -153,6 +157,53 @@ func (c *combinationController) Reconcile(ctx context.Context, req ctrl.Request)
 		Message:            "evaluations successfully processed",
 	})
 
+	// If manifest application is indicated, create generated manifests on-cluster.
+	if combination.Spec.Apply {
+		// update status to indicate processing
+		combination.SetStatusCondition(metav1.Condition{
+			Type:               v1alpha1.TypeInProgress,
+			Status:             metav1.ConditionTrue,
+			Reason:             v1alpha1.ReasonProcessing,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+			Message:            "application of evaluations to the cluster in progress",
+		})
+
+		for _, manifest := range generatedManifests {
+			dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(manifest), 10)
+			obj := &unstructured.Unstructured{}
+			err := dec.Decode(obj)
+			if err != nil {
+				combination.SetStatusCondition(metav1.Condition{
+					Type:               v1alpha1.TypeInvalid,
+					Status:             metav1.ConditionFalse,
+					Reason:             v1alpha1.ReasonApplyFailed,
+					LastTransitionTime: metav1.NewTime(time.Now()),
+					Message:            fmt.Sprintf("failed to decode manifest %s to apply to cluster: %s", obj.GetName(), err.Error()),
+				})
+				return reconcile.Result{}, errors.NewAggregate([]error{err, c.Status().Update(ctx, combination)})
+			}
+			err = c.Create(ctx, obj)
+			if err != nil && !k8serrors.IsAlreadyExists(err) {
+				combination.SetStatusCondition(metav1.Condition{
+					Type:               v1alpha1.TypeInvalid,
+					Status:             metav1.ConditionFalse,
+					Reason:             v1alpha1.ReasonApplyFailed,
+					LastTransitionTime: metav1.NewTime(time.Now()),
+					Message:            fmt.Sprintf("failed to apply manifest %s to cluster: %s", obj.GetName(), err.Error()),
+				})
+				return reconcile.Result{}, errors.NewAggregate([]error{err, c.Status().Update(ctx, combination)})
+			}
+		}
+		// update status once all manifests have been applied
+		combination.SetStatusCondition(metav1.Condition{
+			Type:               v1alpha1.TypeFinished,
```

Comment:
If a previous reconcile finished successfully, but a subsequent reconcile fails, shouldn't we set this condition to `Finished` == `False`?